source/AI.cpp

```diff
@@ -835,13 +835,24 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 				it->SetParent(parent);
 			}
 			// Otherwise, check if this ship wants to return to its parent (e.g. to repair).
-			else if(parentHasSpace && ShouldDock(*it, *parent, playerSystem))
+			else if(ShouldDock(*it, *parent, playerSystem))
 			{
 				it->SetTargetShip(parent);
-				MoveTo(*it, command, parent->Position(), parent->Velocity(), 40., .8);
-				command |= Command::BOARD;
-				it->SetCommands(command);
-				continue;
+
+				if(const auto *bay = it->GetBay())
+				{
+					Point exitPoint = parent->Position() + parent->Facing().Rotate(bay->point);
+					if(MoveTo(*it, command, exitPoint, parent->Velocity(), 15., .3))
+						command |= Command::BOARD;
```

Comment:
For the first question it was to prevent the carry docking before being above the bay (because you can board the ship from anywhere).

For the second - no. We want the carry to move to its bay.